cicecore/shared/ice_calendar.F90

```diff
@@ -402,9 +452,31 @@ subroutine calendar()
 
       ! Restart writing flag
 
+      if (dumpfreq_base == 'zero') then
+         elapsed_years = myear
+         elapsed_months =  elapsed_years * months_per_year + mmonth - 1
+         elapsed_days = compute_days_between(0,1,1,myear,mmonth,mday)
+         elapsed_hours = elapsed_days * hours_per_day + hour
+      elseif (dumpfreq_base == 'init') then
+         elapsed_years = myear - year_init
+         elapsed_months =  elapsed_years * months_per_year + (mmonth - month_init)
+         elapsed_days = compute_days_between(year_init,month_init,day_init,myear,mmonth,mday)
+         elapsed_hours = elapsed_days * hours_per_day + hour
+      else
+         write(nu_diag,*) trim(subname),' ERROR dumpfreq_base not recognized, ',trim(dumpfreq_base)
+         call abort_ice(subname//'ERROR: dumpfreq_base value invalid')
+      endif
```

Comment:
I think it would be cleaner to have a separate subroutine that computes `elapsed_{years,months,days,hours}` depending on the 'init' and 'zero' setting, and have separate variables for the restart and history parts of the subroutine. This way the computation themselves are not repeated in the code.

Reply:
Good point.  I don't know if we need different variables but creating a new subroutine is a good idea.

Reply:
I implemented a new subroutine to improve code reuse.